backends/azure/azure.go

```diff
@@ -0,0 +1,495 @@
+		opt.EndpointURL = DefaultEndpointURL
+	}
+	if opt.Concurrency == 0 {
+		opt.Concurrency = DefaultConcurrency
+	}
+
+	if err := opt.Check(); err != nil {
+		return nil, err
+	}
+
+	log := opt.Logger
+	if log.GetSink() == nil {
+		log = logr.Discard()
+	}
+	log = log.WithName("azure")
+
+	var endpoint string
+
+	accountName := opt.AccountName
+
+	if opt.EndpointURL != "" {
+		endpoint = opt.EndpointURL
+	} else {
+		endpoint = fmt.Sprintf("https://%s.blob.core.windows.net", accountName)
+	}
+
+	var client *azblob.Client
+
+	// If UseEnvCreds is set, we will attempt to use the environment variables and the Azure service principle based `azidentity.NewDefaultAzureCredential()` method
+	// https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/README.md#service-principal-with-secret
+	if opt.UseEnvCreds {
+		// Test if the environment variables are set
+		_, ok := os.LookupEnv("AZURE_CLIENT_ID")
+		if !ok {
+			return nil, errors.New("AZURE_CLIENT_ID could not be found")
+		}
+
+		_, ok = os.LookupEnv("AZURE_TENANT_ID")
+		if !ok {
+			return nil, errors.New("AZURE_TENANT_ID could not be found")
+		}
+
+		_, ok = os.LookupEnv("AZURE_CLIENT_SECRET")
+		if !ok {
+			return nil, errors.New("AZURE_CLIENT_SECRET could not be found")
+		}
+
+		cred, err := azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return nil, err
+		}
+
+		client, err = azblob.NewClient(endpoint, cred, nil)
+
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		cred, err := azblob.NewSharedKeyCredential(accountName, opt.AccountKey)
+		if err != nil {
+			return nil, err
+		}
+
+		client, err = azblob.NewClientWithSharedKeyCredential(endpoint, cred, nil)
+
+		if err != nil {
+			return nil, err
+		}
+	}
+	if opt.CreateContainer {
+		// Create bucket if it does not exist
+		metricCalls.WithLabelValues("create-container").Inc()
+		metricLastCallTimestamp.WithLabelValues("create-container").SetToCurrentTime()
+
+		_, err := client.CreateContainer(context.TODO(), opt.Container, &azblob.CreateContainerOptions{
+			Metadata: map[string]*string{"hello": to.Ptr("world")},
+		})
+
+		if err != nil {
+			if strings.Contains(err.Error(), "ContainerAlreadyExists") {
+				logrus.WithField("storage_type", "Azure").Infof("Container already exists: %s", opt.Container)
+			} else {
+				return nil, err
+			}
+		}
+	}
+
+	b := &Backend{
+		opt:    opt,
+		client: client,
+		log:    log,
+	}
+
+	b.setGlobalPrefix(opt.GlobalPrefix)
+
+	return b, nil
+}
+
+func (b *Backend) List(ctx context.Context, prefix string) (blobList simpleblob.BlobList, err error) {
+	// Handle global prefix
+	combinedPrefix := b.prependGlobalPrefix(prefix)
+
+	if !b.opt.UseUpdateMarker {
+		return b.doList(ctx, combinedPrefix)
+	}
+
+	// Using Load, that will itself prepend the global prefix to the marker name.
+	// So we're using the raw marker name here.
+	m, err := b.Load(ctx, UpdateMarkerFilename)
+
+	var notFound bool
+
+	if err != nil {
+		notFound = err.Error() == "BlobNotFound"
```

Comment:
`bloberror.HasCode` seems more robust to perform this check, could you please use it?